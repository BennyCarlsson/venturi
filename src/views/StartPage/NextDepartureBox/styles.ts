import styled from 'styled-components';
import colors from 'tokens/colors';
import { ReactComponent as TrainLogo } from 'assets/icons/train.svg';
import Paragraph from 'components/Paragraph';

export const NextDepartureWrapper = styled.div`
  border-radius: 0 0 8px 8px;
  background-color: ${colors.vBlue};
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
`;

export const TopPart = styled.div`
  padding: 24px;
  border-bottom: 1px solid ${colors.borderWhite};
`;

export const BottomPart = styled.div`
  display: flex;
`;

export const BottomLeftPart = styled.div`
  padding: 16px 24px;
  width: 100%;
  border-right: 1px solid ${colors.borderWhite};
`;

export const BottomRightPart = styled.div`
  padding: 16px 24px;
  width: 100%;
`;

export const BottomParagraph = styled(Paragraph)`
  padding-top: 8px;
`;

export const StyledTrailLogo = styled(TrainLogo)`
  width: 32px;
  height: 32px;
  padding-right: 16px;
  path {
    fill: ${colors.white};
  }
`;

export const DepartsWrapper = styled.div`
  display: flex;
  padding-top: 8px;
`;
