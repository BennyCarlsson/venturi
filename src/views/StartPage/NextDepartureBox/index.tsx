import Paragraph, { fontWeight } from 'components/Paragraph';
import colors from 'tokens/colors';
import { Departure } from 'types';
import { Box, Heading, Heading2, Heading3, Heading4 } from './styles';

const NextDepartureBox = (props: Departure) => {
  const { stop, time, rtTime, direction, sname } = props;
  return (
    <Box>
      <Heading>Departs</Heading>
      <Heading2>{stop}</Heading2>
      <Heading3>{time}</Heading3>
      <Heading4>{rtTime}</Heading4>
      <Paragraph
        color={colors.vGray}
        fontSize={14}
        fontWeight={fontWeight.normal}
      >{`${sname} ${direction}`}</Paragraph>
    </Box>
  );
};

export default NextDepartureBox;
