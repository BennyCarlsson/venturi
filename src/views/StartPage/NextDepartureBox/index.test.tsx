import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NextDepartureBox from '.';

describe('<Documentation />', () => {
  const name = 'Test name';
  const track = 'Test Track';

  describe('snapshots', () => {
    it('should match snapshot', () => {
      const tree = renderer.create(<NextDepartureBox name={name} track={track} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Component', () => {
    beforeEach(() => {
      render(<NextDepartureBox name={name} track={track} />);
    });

    it('should render without crashing', () => {
      expect(screen).toBeTruthy();
    });

    it('should render name and track props', () => {
      expect(screen.getByText(name)).toBeInTheDocument();
      expect(screen.getByText(track)).toBeInTheDocument();
    });
  });
});
