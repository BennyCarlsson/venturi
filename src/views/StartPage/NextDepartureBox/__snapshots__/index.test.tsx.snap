// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Documentation /> snapshots should match snapshot 1`] = `
<div
  className="sc-iBPRYJ HCnjh"
>
  <div
    className="sc-fubCfw hIKVhW"
  >
    <h1
      className="sc-bdfBwQ UZuHf"
      color="#FFFFFF"
      fontSize={18}
      fontWeight={500}
    >
      Departs from
    </h1>
    <div
      className="sc-dQppl dWSjyx"
    >
      <svg
        className="sc-crrsfI fCASlj"
      >
        train.svg
      </svg>
      <h2
        className="sc-gsTCUz htLLms"
        color="#FFFFFF"
        fontSize={28}
        fontWeight={500}
      >
        Test name
      </h2>
    </div>
  </div>
  <div
    className="sc-pFZIQ hxExeM"
  >
    <div
      className="sc-jrAGrp jzTLuk"
    >
      <p
        className="sc-gKsewC bMoIAp"
        color="#FFFFFF"
        fontSize={14}
        fontWeight={500}
      >
        Platform
      </p>
      <p
        className="sc-gKsewC iTDTZO sc-iqHYGH jyHRlY"
        color="#FFFFFF"
        fontSize={24}
        fontWeight={500}
      >
        Test Track
      </p>
    </div>
    <div
      className="sc-kEjbxe bnNZBb"
    >
      <p
        className="sc-gKsewC bMoIAp"
        color="#FFFFFF"
        fontSize={14}
        fontWeight={500}
      >
        Destination name
      </p>
      <p
        className="sc-gKsewC iTDTZO sc-iqHYGH jyHRlY"
        color="#FFFFFF"
        fontSize={24}
        fontWeight={500}
      >
        Placeholder
      </p>
    </div>
  </div>
</div>
`;
